### PR TITLE
feat: implement rego from dataflow

### PR DESCRIPTION
**Pull request by vjerci**
## Description
This pr adds boilerplate for policies and implements first logger leaks policy.

Example output:
```
- level: warning
  locations:
    - filename: logging.rb
      line_number: "15"
    - filename: logging.rb
      line_number: "21"
    - filename: logging.rb
      line_number: "23"
```

## Checklist

- [x] I've added test coverage that shows my fix or feature works as expected.
- [x] I've updated or added documentation if required.
- [x] I've included usage information in the description if CLI behavior was updated or added.
- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

File: pkg/classification/dependencies/dependencies.go [hunk 1]
```diff
@@ -15,9 +15,9 @@ type ClassifiedDependency struct {
 }
 
 type Classification struct {
-	RecipeMatch bool   `json:"recipe_match"`
-	RecipeName  string `json:"recipe_name,omitempty"`
-	Decision    classify.ClassificationDecision
+	RecipeMatch bool                            `json:"recipe_match"`
+	RecipeName  string                          `json:"recipe_name,omitempty"`
+	Decision    classify.ClassificationDecision `json:"descision"`
```
Comment (cfabianski) [inline pkg/classification/dependencies/dependencies.go]:
```suggestion
	Decision    classify.ClassificationDecision `json:"decision"`
```

File: pkg/flag/report_flags.go [hunk 2]
```diff
@@ -22,7 +23,7 @@ var (
 		Name:       "report",
 		ConfigName: "report.report",
 		Value:      ReportDetectors,
-		Usage:      "specify the kind of report (detectors, dataflow)",
+		Usage:      "specify the kind of report (detectors, dataflow, policies)",
```
Comment (cfabianski) [inline pkg/flag/report_flags.go]: Surprised you don’t have `stats` here 🤔

File: pkg/report/output/output.go [hunk 3]
```diff
@@ -64,12 +65,34 @@ func getReportOutput(report types.Report, config settings.Config) (any, error) {
 		}
 
 		ouputDetections, err = dataflow.GetOuput(detections, config)
+		// output, err := policies.GetDataflow(detections)
```
Comment (cfabianski) [inline pkg/report/output/output.go]: ❓

File: pkg/report/output/output.go [hunk 4]
```diff
@@ -64,12 +65,34 @@ func getReportOutput(report types.Report, config settings.Config) (any, error) {
...
 		if err != nil {
 			return nil, err
 		}
+
+		return ouputDetections, nil
```
Comment (cfabianski) [inline pkg/report/output/output.go]:
```suggestion
		return outputDetections, nil
```

File: pkg/report/output/output.go [hunk 5]
```diff
@@ -64,12 +65,34 @@ func getReportOutput(report types.Report, config settings.Config) (any, error) {
...
+			return nil, err
+		}
+
+		data, err := policies.GetPolicies(policiesData, config)
+		if err != nil {
+			return nil, err
+		}
+
+		return data, err
+		// log.Debug().Msgf("%s", data)
```
Comment (cfabianski) [inline pkg/report/output/output.go]: ❓

**Review by elsapet — COMMENTED** [review]
Does it make sense to add some tests at this point, or is this for a later stage?

File: pkg/flag/report_flags.go (hunk 2 above)
Comment (vjerci) [reply pkg/flag/report_flags.go]: I thought you will add stats?

**Comment by vjerci** [comment]
> Does it make sense to add some tests at this point, or is this for a later stage?

in my opinion it is too early.

File: pkg/flag/report_flags.go (hunk 2 above)
Comment (vjerci) [reply pkg/flag/report_flags.go]: merged and added resolving this

File: pkg/commands/process/settings/policies/logger_leaks.rego [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+import future.keywords
+
+default level := "none"
+
+
+locations[location] {
+    some detector in input.risks
+    detector.detector_id == "detect_ruby_logger"
+    location = detector.data_types[_].locations[_]
```
Comment (cfabianski) [inline pkg/commands/process/settings/policies/logger_leaks.rego:11]: Could you do something to show the level changing depending on the criticality of the information passed?

File: pkg/commands/process/settings/policies.yml [hunk 7]
```diff
@@ -0,0 +1,7 @@
+logger_leaks:
+  message: "Logger leaks detected"
+  level: "warning"
+  query: "level = data.bearer.logger_leaks.level; locations = data.bearer.logger_leaks.locations"
```
Comment (cfabianski) [inline pkg/commands/process/settings/policies.yml]: Could you explain a bit why we have 2 files for 1 policy? 
The `YAML` and the `.rego`

Comment (vjerci) [reply pkg/commands/process/settings/policies.yml]: discussed on the call, but as summary:
it is requirement of underlying opa/rego library.
as discussed query is now on 2 lines

**Review by cfabianski — APPROVED** [review]

**Comment by cfabianski** [comment]
```yaml
- detector_id: logger_leak
  level: warning
  data_types:
    - name: Email
      locations:
       - filename: logging.rb
         line_number: "15"
       - filename: logging.rb
         line_number: "21"
       - filename: logging.rb
         line_number: "23"
```

**Pull request merged**